apis/actions.github.com/v1alpha1/autoscalingrunnerset_types.go

```diff
@@ -80,7 +81,39 @@ type AutoscalingRunnerSetSpec struct {
 
 type GitHubServerTLSConfig struct {
 	// Required
-	RootCAsConfigMapRef string `json:"certConfigMapRef,omitempty"`
+	CertificateFrom *TLSCertificateSource `json:"certificateFrom,omitempty"`
+}
+
+func (c *GitHubServerTLSConfig) ToCertPool(keyFetcher func(name, key string) ([]byte, error)) (*x509.CertPool, error) {
+	if c.CertificateFrom == nil {
+		return nil, fmt.Errorf("certificateFrom not specified")
+	}
+
+	if c.CertificateFrom.ConfigMapKeyRef == nil {
+		return nil, fmt.Errorf("configMapKeyRef not specified")
+	}
+
+	cert, err := keyFetcher(c.CertificateFrom.ConfigMapKeyRef.Name, c.CertificateFrom.ConfigMapKeyRef.Key)
+	if err != nil {
+		return nil, fmt.Errorf(
+			"failed to fetch key %s in %s: %w",
```

Comment:
```suggestion
			"failed to fetch key %q with name %q: %w",
```
nit
It might be cleaner to see: 
```bash
failed to fetch key "my-key" with name "my-cm": ...
```

Reply:
ignore if you think this is not useful :smile:

Reply:
+1 for `%q`, not sure I get `with name`? `in configmap %q` perhaps?